src/material-experimental/mdc-slider/slider.ts

```diff
@@ -20,18 +20,143 @@ import {
-import {SliderAdapter} from './slider-adapter';
-import {MatSliderThumb} from './slider-thumb';
+import {SpecificEventListener, EventType} from '@material/base';
+import {MDCSliderAdapter, MDCSliderFoundation, Thumb, TickMark} from '@material/slider';
+
+/**
+ * Represents a drag event emitted by the MatSlider component.
+ */
+export interface MatSliderDragEvent {
+  /** The MatSliderThumb that was interacted with. */
+  source: MatSliderThumb;
+
+  /** The MatSlider that was interacted with. */
+  parent: MatSlider;
+
+  /** The current value of the slider. */
+  value: number;
+}
+
+/**
+ * The native input used by the MatSlider.
+ */
+@Directive({
+  selector: 'input[matSliderThumb], input[matSliderStartThumb], input[matSliderEndThumb]',
+  host: {
+    'class': 'mdc-slider__input',
+    'type': 'range',
+    '(blur)': '_blur.emit()',
+    '(focus)': '_focus.emit()',
+  },
+})
+export class MatSliderThumb implements AfterViewInit {
+
+  // ** IMPORTANT NOTE **
+  //
+  // The way `value` is implemented for MatSliderThumb goes against our standard practice. Normally
+  // we would define a private variable `_value` as the source of truth for the value of the slider
+  // thumb input. The source of truth for the value of the slider inputs has already been decided
+  // for us by MDC to be the value attribute on the slider thumb inputs. This is because the MDC
+  // foundation and adapter expect that the value attribute is the source of truth for the slider
+  // inputs.
+  //
+  // Also, note that the value attribute is completely disconnected from the value property.
+
+  /** The current value of this slider input. */
+  @Input()
+  get value(): number {
+    return coerceNumberProperty(this._elementRef.nativeElement.getAttribute('value'));
+  }
+  set value(v: number) {
+    const value = coerceNumberProperty(v);
+
+    // If the foundation has already been initialized, we need to
+    // relay any value updates to it so that it can update the UI.
+    if (this._slider._initialized) {
+      this._slider._setValue(value, this._thumb);
+    } else {
+      // Setup for the MDC foundation.
+      this._elementRef.nativeElement.setAttribute('value', `${value}`);
+    }
+  }
+
+  /** Event emitted when the slider thumb starts being dragged. */
+  @Output() readonly dragStart: EventEmitter<MatSliderDragEvent>
+    = new EventEmitter<MatSliderDragEvent>();
+
+  /** Event emitted when the slider thumb stops being dragged. */
+  @Output() readonly dragEnd: EventEmitter<MatSliderDragEvent>
+    = new EventEmitter<MatSliderDragEvent>();
+
+  /** Event emitted every time the MatSliderThumb is blurred. */
+  @Output() readonly _blur: EventEmitter<void> = new EventEmitter<void>();
+
+  /** Event emitted every time the MatSliderThumb is focused. */
+  @Output() readonly _focus: EventEmitter<void> = new EventEmitter<void>();
+
+  /** Indicates which slider thumb this input corresponds to. */
+  private _thumb: Thumb;
```

Comment:
The `Thumb` enum comes from MDC

Reply:
Ack, I would still change the name of the property

Reply:
Ok. Would you also like me to make this change to all of the `SliderAdapter` and `MatSlider` methods that use `thumb` as the name for this?

Reply:
Yeah, any properties/variables that refer to this concept I would name `thumbPosition` since it's a much more clear description than `thumb` (which I would assume is a reference to the actual thumb itself, not just the position)